docs/examples/use_cases/tensorflow/efficientdet/train.py

```diff
@@ -0,0 +1,239 @@
+            validation_data=eval_dataset if args.eval_during_training else None,
+            validation_steps=args.eval_steps,
+            validation_freq=args.eval_freq,
+        )
+
+        if args.eval_after_training:
+            print("Evaluation after training:")
+            model.evaluate(eval_dataset, steps=args.eval_steps)
+
+        model.save_weights(args.output_filename)
+
+
+if __name__ == "__main__":
+    import argparse
+    from argparse_utils import enum_action
+
+    parser = argparse.ArgumentParser(formatter_class=utils.SmartFormatter)
+    parser.add_argument(
+        "--initial_epoch",
+        type=int,
+        default=0,
+        help="Epoch from which to start training.",
+    )
+    parser.add_argument(
+        "--epochs", type=int, default=300, help="Epoch on which training should finish."
+    )
+    parser.add_argument(
+        "--input_type",
+        action=enum_action(utils.InputType),
+        required=True,
+        help="Input type.",
+    )
+    parser.add_argument(
+        "--images_path",
+        help="Path to COCO images.",
+    )
+    parser.add_argument(
+        "--annotations_path",
+        help="Path to COCO annotations.",
+    )
+    parser.add_argument(
+        "--train_file_pattern",
+        help="TFrecord files glob pattern for files with training data.",
+    )
+    parser.add_argument("--batch_size", type=int, default=64)
+    parser.add_argument(
+        "--train_steps",
+        type=int,
+        default=2000,
+        help="Number of steps (iterations) in each epoch.",
+    )
+    parser.add_argument(
+        "--eval_file_pattern",
+        help="TFrecord files glob pattern for files with evaluation data, "
+        "defaults to `train_file_pattern` if not given.",
+    )
+    parser.add_argument(
+        "--eval_steps",
+        type=int,
+        default=5000,
+        help="Number of examples to evaluate during each evaluation.",
+    )
+    parser.add_argument(
+        "--eval_freq", type=int, default=1, help="During training evaluation frequency."
+    )
+    parser.add_argument(
+        "--eval_during_training",
+        action="store_true",
+        help="Whether to run evaluation every `eval_freq` epochs.",
+    )
+    parser.add_argument(
+        "--eval_after_training",
+        action="store_true",
+        help="Whether to run evaluation after finished training.",
+    )
+    parser.add_argument(
+        "--pipeline_type",
+        action=enum_action(utils.PipelineType),
+        required=True,
+        help="R|Pipeline type used while loading and preprocessing data. One of:\n"
+        "tensorflow – pipeline used in original EfficientDet implementation on https://github.com/google/automl/tree/master/efficientdet;\n"
+        "synthetic – like `tensorflow` pipeline type but repeats one batch endlessly;\n"
+        "dali_gpu – pipeline which uses Nvidia Data Loading Library (DALI) to run part of data preprocessing on GPUs to improve efficiency;\n"
```

Comment:
```suggestion
        "dali_gpu – pipeline which uses NVIDIA DALI to run part of data preprocessing on GPUs to improve efficiency;\n"
```
I'd go with this, but it's a matter of taste. Anyway, `NVIDIA` should be capitalised

Reply:
Done